go/vt/vtgate/planbuilder/operators/fk_cascade.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operators
+
+import (
+	"vitess.io/vitess/go/vt/vtgate/planbuilder/operators/ops"
+)
+
+// FkCascade is used to represent a foreign key cascade operation
+// as an operator. This operator is created for DML queries that require
+// cascades (for example, ON DELETE CASCADE).
+type FkCascade struct {
+	Selection ops.Operator
+	Children  []*FkChild
```

Comment:
not clear why we need the FkChild struct here. why can't these be just `Operator`?

Reply:
We need to set the bind variable names in the FkChild structs and also the column indexes. Also, for `FkCascade` it doesn't make sense to have the Children as any other operator.
We could store them as `ops.Operator`s too, but would require typecasting checks everywhere we access them.

Reply:
I'm not following. are you talking about future code? I can't find any instance where we touch the innards of FkChild after creation